app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -124,6 +135,43 @@ def get_images
         process_collection(images, :vms) { |image| parse_image(image) }
       end
 
+      def get_security_groups
+        security_groups = gather_data_for_this_region(@nsg)
+
+        process_collection(security_groups, :security_groups) do |sg|
+          parse_security_group(sg)
+        end
+
+        get_firewall_rules
+      end
+
+      # To get the list of security groups for a VM we need to find any
+      # NIC's associated with the security group, and then get the VM's
+      # associated with the NIC's.
+      #
+      def get_security_groups_for_vm(vm)
+        group_array = []
+
+        security_groups.each do |sg|
+          sg.properties.network_interfaces.each do |nic|
+            resource_group = nic.id[%r{resourceGroups/(.*?)/}, 1]
+            nic = @nis.get(File.basename(nic.id), resource_group)
+            if nic.properties['virtualMachine'] and nic.properties.virtual_machine.id == vm.id
+              group_array << sg
+            end
+          end
+        end
+
+        group_array
+      end
+
+      def get_firewall_rules
+        security_groups.each do |sg|
+          new_sg = @data_index.fetch_path(:security_groups, sg.id)
+          new_sg[:firewall_rules] = parse_firewall_rules(sg)
+        end
+      end
+
```

Comment:
@djberg96 
Can we deal with the firewall rules for a security group in the parse_security_group method instead? If so, it would be more efficient and straightforward.